GetStream/stream-chat-swift
# Add support for automatic translations

### 🔗 Issue Links
Resolves https://github.com/GetStream/ios-issues-tracking/issues/602

### 🎯 Goal
Add support for automatic translation when receiving messages; the user has the language set when connecting a user.

### 📝 Summary
- Allow providing a language when connecting a user
- Allow enabling automatic translation through `Components.default.messageAutoTranslationEnabled`
- Parse the language information for each user
- Parse the original language from each message

### 🛠 Implementation
A new option, `.translation`, was added to `ChatMessageLayoutOption`, which is populated if the automatic translation is enabled and if a translation exists and is valid.

### 🎨 Showcase
<img src="https://github.com/GetStream/stream-chat-swift/assets/12814114/9f32c0e4-be29-4cec-817e-43f0d6b06bf2" width="30%"/>

### 🧪 Manual Testing Notes
Note: Automatic translation is enabled by default on the DemoApp

1. Open the DemoApp
2. Go to configuration in the initial screen
3. Set a language for the user
4. Open a Channel which both users have their language set (ex 1-1 Luke vs Han Solo)
5. Send a message from the participant device
6. A "translated to" label should appear
7. If the message is not translatable, it should not appear that it was translated

### ☑️ Contributor Checklist
